### ISSUE TITLE
[credentials] Test Python 3.11 and 3.12

### ISSUE THREAD

**Issue by feanil** [openedx/credentials#2427]

**Comment by justinhynes** [comment]
I noticed there was a branch in Credentials to add Python 3.11 and 3.12 support, but it doesn't look ready for consumption yet. The branch is still in `draft` status with a few conflicts, we will keep our eyes on this and run through some tests once it looks like the work is ready.

**Comment by DawoudSheraz** [comment]
FYI - Tutor-credentials is blocked on this https://github.com/overhangio/tutor-credentials/issues/34#issuecomment-1993975836

**Comment by feanil** [comment]
@DawoudSheraz why do you need to land the tutor change before landing the update in credentials master?  Or is credentials already testing with python 3.12?

**Comment by DawoudSheraz** [comment]
> @DawoudSheraz why do you need to land the tutor change before landing the update in credentials master? Or is credentials already testing with python 3.12?

We are blocked on credentials upgrade. We did the initial testing of tutor-credentials on Python 3.12 but it failed due to dependencies issues in credentails (see https://github.com/overhangio/tutor-credentials/issues/34#issuecomment-1993975836).

**Comment by kdmccormick** [comment]
As Dawoud mentioned, it seems like 3.12 was blocked on a dependency conflict (didkit) in this repo, not a downstream Tutor issue.

I've updated the title to reflect that we only added 3.11 support. I recommend opening a new issue for 3.12 support, which we'll need in Sumac.
